src/beast/beast/http/README.md

```diff
@@ -0,0 +1,401 @@
+# Beast.HTTP
+
+--------------------------------------------------------------------------------
+
+Beast.HTTP offers programmers simple and performant models of HTTP messages and
+their associated operations including synchronous and asynchronous reading and
+writing using Boost.Asio.
+
+## Introduction
+
+The HTTP protocol is pervasive in network applications. As C++ is a logical
+choice for high performance network servers, there is great utility in solid
+building blocks for manipulating, sending, and receiving HTTP messages compliant
+with RFC2616 and its supplements that follow. Unfortunately popular libraries
+such as Boost or the C++ standard library do not provide support for this
+popular protocol.
+
+Beast.HTTP is built on Boost.Asio and uses HTTP parser from NodeJS, which is
+extensively field tested and exceptionally robust. A proposal to add networking
+functionality to the C++ standard library, based on Boost.Asio, is under
+consideration by the standards committee. Since the final approved networking
+interface for the C++ standard library will likely closely resemble the current
+interface of Boost.Asio, it is logical for Beast.HTTP to use Boost.Asio as its
+network transport.
+
+Beast.HTTP addresses the following goals:
+
+* **Ease of Use:** HTTP messages are modeled using simple, readily
+accessible objects.
+
+* **Flexibility:** The modeling of the HTTP message should allow for
+multiple implementation strategies for representing the content-body.
+
+* **Performance:** The implementation should run sufficiently fast as
+to make it a competitive choice for building a high performance network
+server.
+
+* **Scalability:** The library should facilitate the development of
+network applications that scale to thousands of concurrent connections.
+
+## Example
+
+All examples and identifiers mentioned in this document are written as
+if the following declarations are in effect:
+```C++
+#include <beast/http.h>
+using namespace beast;
+using namespace boost::asio;
+```
+
+Create a HTTP request:
+```C++
+request<string_body> req(method_t::http_get, "/", 11);
+req.headers.insert("Host", "127.0.0.1:80");
+req.headers.insert("User-Agent", "Beast.HTTP");
+
+```
+
+To send a message it must first be prepared through a call to `prepare`. This
+customization point transforms the `message` into a `prepared_message`,
+filling in some standard HTTP behavior and allowing the Body associated with
+the message to perform preparatory steps. For example, a string body may set
+the Content-Length and Content-Type appropriately.
+```C++
+void send_request(ip::tcp::socket& sock,
+    request<string_body>&& req)
+{
+    // Send the request on the socket
+    write(sock, prepare(req, connection(keep_alive));
```

Comment:
After searching for `prepare` and getting 154 hits with many uses not having two parameters, I think the name of the customization point is too confused with other uses of the same name.  I would either rename the customization point, or rename the other uses of `prepare` which aren't customization points.  As it is, I'm not yet clear what the signature and use of the customization point is because of all the false positives in my search for it.